src/store/getters.js

```diff
@@ -11,33 +10,41 @@ import {
   PROTOCOL_AETERNITY,
   TX_DIRECTION,
 } from '@/constants';
-import {
-  getHdWalletAccount,
-} from '@/popup/utils';
 import { useAeSdk } from '@/composables';
 import { AE_SYMBOL } from '@/protocols/aeternity/config';
 import {
   aettosToAe,
   categorizeContractCallTxObject,
 } from '@/protocols/aeternity/helpers';
+import { ProtocolAdapterFactory } from '@/lib/ProtocolAdapterFactory';
 
 export default {
-  wallet({ mnemonic }) {
+  wallet: ({ mnemonic }) => (protocol = PROTOCOL_AETERNITY) => {
```

Comment:
Why not to pass the `protocol` as a second property of the base function (next to the `mnemonic`)?

Reply:
https://vuex.vuejs.org/guide/getters.html#method-style-access
The first parameter is the state.

Reply:
No longer needed.